### PR TITLE
kconfig: generate Kconfig.modules from west list

### PR DESCRIPTION
**Pull request by nashif**
Get a list of modules and put them in a file that is parsed by Kconfig.

This is needed for verifying multi-repo configuration where some Kconfig
options are only defined in module git repo but not in the main tree.

### CODE REVIEW

**Comment by nashif** [comment]
> Issue #1: Should this be common code in CI and in regular builds?

where else is this needed? Why would you need this in regular builds?

File: scripts/check_compliance.py [hunk 1]
```diff
@@ -47,6 +48,29 @@ def get_shas(refspec):
...
+    Get a list of modules and put them in a file that is parsed by Kconfig
+
+    This is needed to complete Kconfig sanity tests.
+
+    """
+    modules = sh.west("list", "--format={posixpath}", **sh_special_args)
+    with open(modules_file, 'w') as km:
+        for m in modules:
+            module_path = m.strip()
+            module_file = "{}/zephyr/module.yml".format(module_path)
```
Comment (carlescufi) [inline scripts/check_compliance.py]: Please use `os.path.join`. Windows paths are different

File: scripts/check_compliance.py [hunk 2]
```diff
@@ -224,7 +248,7 @@ def parse_kconfig(self):
         os.environ['GENERATED_DTS_BOARD_CONF'] = "dummy"
 
         # For multi repo support
-        open(os.path.join(tempfile.gettempdir(), "Kconfig.modules"), 'a').close()
+        get_modules(os.path.join(tempfile.gettempdir(), "Kconfig.modules"))
```
Comment (carlescufi) [inline scripts/check_compliance.py]: How does the root Kconfig.zephyr file find the location of this tempfile?

**Comment by carlescufi** [comment]
> > Issue #1: Should this be common code in CI and in regular builds?
> 
> where else is this needed? Why would you need this in regular builds?

I mean that we now have 2 implementations of this. One in CMake and this one in Python. Would it be better to move this Python code to the zephyr repo and reuse it here? Then we implement it only once, in Python, and remove the current CMake implementation

File: scripts/check_compliance.py (hunk 2 above)
Comment (nashif) [reply scripts/check_compliance.py]: look a few lines up, it is in 

`$(PROJECT_BINARY_DIR)/Kconfig.modules` which translates to 

`os.path.join(tempfile.gettempdir(), "Kconfig.modules")`

**Comment by nashif** [comment]
> I mean that we now have 2 implementations of this. One in CMake and this one in Python. Would it be better to move this Python code to the zephyr repo and reuse it here? Then we implement it only once, in Python, and remove the current CMake implementation

We could, but I would prefer for this to be implemented in the Zephyr tree, get finalised and tested then we can look into how to use it in CI, probably something for after 1.14. This is now resolve a CI issue an can always be optimised when we have a better way for doing this using code from the tree.

File: scripts/check_compliance.py (hunk 2 above)
Comment (carlescufi) [reply scripts/check_compliance.py]: Oh right. I thought that gettempdir() returned a different one every time.

**Comment by marc-hb** [comment]
Fixed crash in PR #36

**Pull request merged**